graphing/src/cloudevolution/views.py

```diff
@@ -139,12 +190,51 @@ def dilutions(request, experiment):
 		# All vials were chemostats or not used
 		efficiency = None
 
+	# Calculate consumption of last bottle
+	bottles = []
+	bottle_info = []  # Stores info displayed in "See bottle setup"
+	bottle_data = open(bottle_file).readlines()[1:]
+
+	# Get experiment start time
+	with open(expt_pickle, 'rb') as f:
+		expt_start = pickle.load(f)[0]
+
+	if not bottle_data:
+		bottle_info = None
+	else:
+		# bottleID	vials	volume0	timestamp0 ... volumeN	timestampN
```

Comment:
So this file can have arbitrary columns - they just get additional columns added every time the user changes the bottle? I think having the file be row based is better, new events just require adding a row as opposed to essentially re-writing the entire file when you update a bottle at the top of the file. Also you will end up with lots of blanks for bottles that don't get replaced at the same rates if you maintain column structure. 

Since everything is timestamped you would just find the latest event with that. Then this file becomes more of a log of bottle changes vs a database type table. Happy to hear your thoughts though if you disagree.